internal/impl/elasticsearch/integration_v2_test.go

```diff
@@ -0,0 +1,269 @@
+		integration.StreamTestOutputOnlySendBatch(10, queryGetFn),
+	)
+	suite.Run(
+		t, template,
+		integration.StreamTestOptPort(resource.GetPort("9200/tcp")),
+	)
+}
+
+func TestIntegrationV2Connect(t *testing.T) {
+	integration.CheckSkip(t)
+	tests := []struct {
+		name       string
+		env        []string
+		pollURL    func(port string) string
+		configYAML func(argsString []string) string
+	}{
+		{
+			name: "NoAuth",
+			env: []string{
+				"discovery.type=single-node",
+				"xpack.security.enabled=false",
+				"xpack.security.transport.ssl.enabled=false",
+				"xpack.security.http.ssl.enabled=false",
+				"ES_JAVA_OPTS=-Xms512m -Xmx512m",
+			},
+			pollURL: func(port string) string {
+				return fmt.Sprintf("http://localhost:%s", port)
+			},
+			configYAML: func(argsString []string) string {
+				argsAny := make([]any, 0, 2)
+				for _, v := range argsString {
+					argsAny = append(argsAny, v)
+				}
+				return fmt.Sprintf(`
+urls:
+  - http://localhost:%s
+index: test-index
+id: ${! uuid_v4() }
+`, argsAny...)
+			},
+		},
+		{
+			name: "BasicAuth",
+			env: []string{
+				"discovery.type=single-node",
+				"xpack.security.enabled=true",
+				"xpack.security.transport.ssl.enabled=false",
+				"xpack.security.http.ssl.enabled=false",
+				"ES_JAVA_OPTS=-Xms512m -Xmx512m",
+				"ELASTIC_PASSWORD=password",
+			},
+			pollURL: func(port string) string {
+				return fmt.Sprintf("http://elastic:password@localhost:%s", port)
+			},
+			configYAML: func(argsString []string) string {
+				argsAny := make([]any, 0, 2)
+				for _, v := range argsString {
+					argsAny = append(argsAny, v)
+				}
+				return fmt.Sprintf(`
+urls: 
+  - http://localhost:%s
+index: test-index
+id: ${! uuid_v4 }
+basic_auth:
+  enabled: true
+  username: elastic
+  password: password
+`, argsAny...)
+			},
+		},
+		{
+			name: "TLS",
+			env: []string{
+				"discovery.type=single-node",
+				"ES_JAVA_OPTS=-Xms512m -Xmx512m",
+				"ELASTIC_PASSWORD=password",
+			},
+			pollURL: func(port string) string {
+				return fmt.Sprintf("https://elastic:password@localhost:%s", port)
+			},
+			configYAML: func(argsString []string) string {
+				argsAny := make([]any, 0, 2)
+				for _, v := range argsString {
+					argsAny = append(argsAny, v)
+				}
+				return fmt.Sprintf(`
+urls:
+  - https://localhost:%s
+index: test-index
+id: ${! uuid_v4 }
+basic_auth:
+  enabled: true
+  username: elastic
+  password: password
+tls:
+  enabled: true
+  root_cas_file: "%s"
+`, argsAny...)
+			},
+		},
+	}
+
+	for _, tc := range tests {
+
+		pool, resource, port := startElasticsearch(t, tc.env)
+		pollURL := tc.pollURL(port)
+
+		var client *http.Client
+		var configArgs []string
+		if tc.name == "TLS" {
+			// give elasticsearch a moment to create a tls cert
+			time.Sleep(time.Second * 20)
+
+			containerName := resource.Container.Name[1:]
+			cmd := exec.Command("docker", "exec", containerName, "cat", "config/certs/http_ca.crt")
```

Comment:
Could be replaced with:

```suggestion
			resource.Exec([]string{"cat", "config/certs/http_ca.crt"}, dockertest.ExecOptions{})
```

Just give a look at whether `ExecOptions` needs to be configured for stdout.